src/test/compile-fail/issue-11004.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::mem;
+
+struct A { x: i32, y: f64 }
+
+#[cfg(not(works))]
+unsafe fn access(n:*mut A) -> (i32, f64) {
+    let x : i32 = n.x; //~ ERROR attempted access of field `x`
+                       //~| NOTE `A` is a native pointer; perhaps you need to deref with `(*A).x`
```

Comment:
Surely `A` isn't the native pointer, `n` is. Also `(*A).x` isn't a valid expression. I think this should be either "`n` is a native pointer; perhaps you need to deref with `(*n).x`" or "`*mut A` is a native pointer; perhaps you need to deref with `(*n).x`".